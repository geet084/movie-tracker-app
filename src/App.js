import React, { Component } from 'react';
import './App.css';
// import { apiKey } from './api-key';

class App extends Component {
  render() {
    return (
      <div className="App">
        <h1>Movie Tracker</h1>
      </div>
    );
  }
}

export default App;
