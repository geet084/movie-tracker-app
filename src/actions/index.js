export const updatePageToDisplay = (page) => ({
  type: 'UPDATE_PAGE_TO_DISPLAY',
  page,
});

export const updateUser = ({ email, password }) => ({
  type: 'UPDATE_USER',
  email,
  password,
});

export const toggleFavorite = (id) => ({
  type: 'TOGGLE_FAVORITE',
  id,
});