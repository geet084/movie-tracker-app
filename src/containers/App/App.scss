header {
  padding: 40px 6%;
  display: flex;
  justify-content: space-between;
  align-items: center;
}

.nav-links {
  color: black;
  font-size: 20px;
  margin: 10px;
  text-decoration: none;
}

.selected {
  color: hotpink;
}

#sign-out-button {
  font-size: 20px;
  border: none;
  background: none;

  &:hover {
    cursor: pointer;
  }

  &:focus {
    outline: none;
  }
}