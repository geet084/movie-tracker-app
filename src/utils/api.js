const fetchData = async (url) => {
  const response = await fetch(url);
  if (response.ok) {
    return response.json();
  } else {
    throw Error(`Error fetching data: ${response.error.message}`);
  }
}

const postData = async (user, root) => {
  const url = `http://localhost:3000/api/users${root}`;
  const response = await fetch(url, {
    method: 'POST',
    body: JSON.stringify(user),
    headers: {
      'Content-type': 'application/json'
    }
  })
  console.log(response)
  if (response.ok) {
    return await response.json()
  } else {
    throw Error(`Error posting data: ${response.error.message}`);
  }
}

export default {
  fetchData,
  postData,
}
