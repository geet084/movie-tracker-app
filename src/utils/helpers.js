import React from 'react'

const buildInput = (type, value, handleChange) => {
  return (
    <label key={type}>{type}
      <input
        required
        name={type}
        type={type}
        onChange={handleChange}
        value={value[type]} />
    </label>
  )
}

export default buildInput