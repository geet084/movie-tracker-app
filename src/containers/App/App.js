import React, { Component } from 'react';
import '../../normalize.css';
import '../../main.scss';
import { connect } from 'react-redux';
import { getMovies } from '../../actions';
import API from '../../utils/api';
import { apiKey } from '../../utils/api-key';

class App extends Component {
  constructor(props) {
    super(props);

  }

  componentDidMount = async () => {
    const url = `https://api.themoviedb.org/3/movie/now_playing?api_key=${apiKey}&language=en-US&page=1`;
    const movies = await API.fetchData(url);
    await this.props.getMovies(movies.results);
  }

  render() {
    return (
      <div className="App">
        <h1>Movie Tracker</h1>
      </div>
    );
  }
}

const mapDispatchToProps = (dispatch) => ({
  getMovies: (movies) => dispatch(getMovies(movies)),
})

export default connect(null, mapDispatchToProps)(App);
