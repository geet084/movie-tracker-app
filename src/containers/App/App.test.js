import React from 'react';
import ReactDOM from 'react-dom';
import { App, mapDispatchToProps } from './App';
import API from '../../utils/api';
import { apiKey } from '../../utils/api-key';
import { shallow } from 'enzyme';
import { getMovies } from '../../actions';

it.skip('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<App />, div);
  ReactDOM.unmountComponentAtNode(div);
});

describe('App', () => {
  describe('defaults', () => {
    let wrapper;

    beforeEach(() => {
      wrapper = shallow(<App />);
    });

    it('should match the snapshot with all data passed in', () => {
      expect(wrapper).toMatchSnapshot();
    });
  });

  describe('componentDidMount', () => {
    let wrapper;
    let mockData;
    let mockGetMovies;

    beforeEach(() => {
      mockGetMovies = jest.fn();
      mockData = { results: [{ name: 'AquaMan', id: 1 }, { name: 'Beauty and the Beast', id: 2 }] };
      API.fetchData = jest.fn().mockImplementation(() => {
        return mockData
      });
      wrapper = shallow(
      <App 
        getMovies={mockGetMovies}
      />);
    });

    it('should call fetchData with the correct parameters', async () => {
      const expected = `https://api.themoviedb.org/3/movie/popular?page=1&api_key=${apiKey}&language=en-US`;
      await wrapper.instance().componentDidMount();
      expect(API.fetchData).toHaveBeenCalledWith(expected);
    });

    it('should call getMovies with the correct parameters', async () => {
      const expected = [{ name: 'AquaMan', id: 1 }, { name: 'Beauty and the Beast', id: 2 }];
      await wrapper.instance().componentDidMount();
      expect(mockGetMovies).toHaveBeenCalledWith(expected);
    });
  });

  describe('mapDispatchToProps', () => {
    it('should call dispatch when using a function from mapDispatchToProps', () => {
      const mockDispatch = jest.fn();
      const actionToDispatch = getMovies([{ name: 'AquaMan', id: 1 }, { name: 'Beauty and the Beast', id: 2 }]);
      const mappedProps = mapDispatchToProps(mockDispatch);
      mappedProps.getMovies([{ name: 'AquaMan', id: 1 }, { name: 'Beauty and the Beast', id: 2 }]);
      expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch);
    });
  });
});