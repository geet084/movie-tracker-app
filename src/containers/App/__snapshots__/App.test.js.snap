// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App defaults should match the snapshot with all data passed in 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App
    user={
      Object {
        "email": "a@a",
        "favorites": Array [],
        "id": 1,
        "name": "Tommy",
        "password": "password",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <header>
          <h1>
            Movie Tracker
          </h1>
          <div>
            <span
              className="user-name"
            >
              Welcome 
              Tommy
            </span>
            <NavLink
              activeClassName="selected"
              aria-current="page"
              className="nav-links"
              to="/"
            >
              Popular Movies
            </NavLink>
            <NavLink
              activeClassName="selected"
              aria-current="page"
              className="nav-links"
              to="/favorites"
            >
              Favorites
            </NavLink>
            <button
              className="nav-links"
              id="sign-out-button"
              onClick={[Function]}
            >
              Sign Out
            </button>
          </div>
        </header>,
        <Switch>
          <Route
            component={[Function]}
            path="/"
          />
          <Route
            path="/favorites"
            render={[Function]}
          />
        </Switch>,
        <Route
          path="/movies/:id"
          render={[Function]}
        />,
        <Route
          component={[Function]}
          path="/login"
        />,
        <Route
          component={[Function]}
          path="/signup"
        />,
      ],
      "className": "App",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h1>
              Movie Tracker
            </h1>,
            <div>
              <span
                className="user-name"
              >
                Welcome 
                Tommy
              </span>
              <NavLink
                activeClassName="selected"
                aria-current="page"
                className="nav-links"
                to="/"
              >
                Popular Movies
              </NavLink>
              <NavLink
                activeClassName="selected"
                aria-current="page"
                className="nav-links"
                to="/favorites"
              >
                Favorites
              </NavLink>
              <button
                className="nav-links"
                id="sign-out-button"
                onClick={[Function]}
              >
                Sign Out
              </button>
            </div>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Movie Tracker",
            },
            "ref": null,
            "rendered": "Movie Tracker",
            "type": "h1",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <span
                  className="user-name"
                >
                  Welcome 
                  Tommy
                </span>,
                <NavLink
                  activeClassName="selected"
                  aria-current="page"
                  className="nav-links"
                  to="/"
                >
                  Popular Movies
                </NavLink>,
                <NavLink
                  activeClassName="selected"
                  aria-current="page"
                  className="nav-links"
                  to="/favorites"
                >
                  Favorites
                </NavLink>,
                <button
                  className="nav-links"
                  id="sign-out-button"
                  onClick={[Function]}
                >
                  Sign Out
                </button>,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    "Welcome ",
                    "Tommy",
                  ],
                  "className": "user-name",
                },
                "ref": null,
                "rendered": Array [
                  "Welcome ",
                  "Tommy",
                ],
                "type": "span",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "activeClassName": "selected",
                  "aria-current": "page",
                  "children": "Popular Movies",
                  "className": "nav-links",
                  "to": "/",
                },
                "ref": null,
                "rendered": "Popular Movies",
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "activeClassName": "selected",
                  "aria-current": "page",
                  "children": "Favorites",
                  "className": "nav-links",
                  "to": "/favorites",
                },
                "ref": null,
                "rendered": "Favorites",
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Sign Out",
                  "className": "nav-links",
                  "id": "sign-out-button",
                  "onClick": [Function],
                },
                "ref": null,
                "rendered": "Sign Out",
                "type": "button",
              },
            ],
            "type": "div",
          },
        ],
        "type": "header",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <Route
              component={[Function]}
              path="/"
            />,
            <Route
              path="/favorites"
              render={[Function]}
            />,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "component": [Function],
              "path": "/",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "path": "/favorites",
              "render": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "path": "/movies/:id",
          "render": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "component": [Function],
          "path": "/login",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "component": [Function],
          "path": "/signup",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <header>
            <h1>
              Movie Tracker
            </h1>
            <div>
              <span
                className="user-name"
              >
                Welcome 
                Tommy
              </span>
              <NavLink
                activeClassName="selected"
                aria-current="page"
                className="nav-links"
                to="/"
              >
                Popular Movies
              </NavLink>
              <NavLink
                activeClassName="selected"
                aria-current="page"
                className="nav-links"
                to="/favorites"
              >
                Favorites
              </NavLink>
              <button
                className="nav-links"
                id="sign-out-button"
                onClick={[Function]}
              >
                Sign Out
              </button>
            </div>
          </header>,
          <Switch>
            <Route
              component={[Function]}
              path="/"
            />
            <Route
              path="/favorites"
              render={[Function]}
            />
          </Switch>,
          <Route
            path="/movies/:id"
            render={[Function]}
          />,
          <Route
            component={[Function]}
            path="/login"
          />,
          <Route
            component={[Function]}
            path="/signup"
          />,
        ],
        "className": "App",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h1>
                Movie Tracker
              </h1>,
              <div>
                <span
                  className="user-name"
                >
                  Welcome 
                  Tommy
                </span>
                <NavLink
                  activeClassName="selected"
                  aria-current="page"
                  className="nav-links"
                  to="/"
                >
                  Popular Movies
                </NavLink>
                <NavLink
                  activeClassName="selected"
                  aria-current="page"
                  className="nav-links"
                  to="/favorites"
                >
                  Favorites
                </NavLink>
                <button
                  className="nav-links"
                  id="sign-out-button"
                  onClick={[Function]}
                >
                  Sign Out
                </button>
              </div>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Movie Tracker",
              },
              "ref": null,
              "rendered": "Movie Tracker",
              "type": "h1",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <span
                    className="user-name"
                  >
                    Welcome 
                    Tommy
                  </span>,
                  <NavLink
                    activeClassName="selected"
                    aria-current="page"
                    className="nav-links"
                    to="/"
                  >
                    Popular Movies
                  </NavLink>,
                  <NavLink
                    activeClassName="selected"
                    aria-current="page"
                    className="nav-links"
                    to="/favorites"
                  >
                    Favorites
                  </NavLink>,
                  <button
                    className="nav-links"
                    id="sign-out-button"
                    onClick={[Function]}
                  >
                    Sign Out
                  </button>,
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      "Welcome ",
                      "Tommy",
                    ],
                    "className": "user-name",
                  },
                  "ref": null,
                  "rendered": Array [
                    "Welcome ",
                    "Tommy",
                  ],
                  "type": "span",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "activeClassName": "selected",
                    "aria-current": "page",
                    "children": "Popular Movies",
                    "className": "nav-links",
                    "to": "/",
                  },
                  "ref": null,
                  "rendered": "Popular Movies",
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "activeClassName": "selected",
                    "aria-current": "page",
                    "children": "Favorites",
                    "className": "nav-links",
                    "to": "/favorites",
                  },
                  "ref": null,
                  "rendered": "Favorites",
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Sign Out",
                    "className": "nav-links",
                    "id": "sign-out-button",
                    "onClick": [Function],
                  },
                  "ref": null,
                  "rendered": "Sign Out",
                  "type": "button",
                },
              ],
              "type": "div",
            },
          ],
          "type": "header",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <Route
                component={[Function]}
                path="/"
              />,
              <Route
                path="/favorites"
                render={[Function]}
              />,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "component": [Function],
                "path": "/",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "path": "/favorites",
                "render": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "path": "/movies/:id",
            "render": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "component": [Function],
            "path": "/login",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "component": [Function],
            "path": "/signup",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
